File: go/internal/sol/p0007/p0007.go
package p0007

import (
	"strconv"

	"pe-solver/internal/mylib"
)

func compute(nth uint) string {
	if nth == 0 {
		panic("The argument must be positive")
	}

	pGen := mylib.PrimeGenerator{}

	// Drop 'nth - 1' prime numbers
	for range nth - 1 {
		_ = pGen.Next()
	}

	return strconv.FormatUint(uint64(pGen.Next()), 10)
}

func Solve() string {
	return compute(10_001)
}

File: haskell/src/Sol/P0007.hs
module Sol.P0007 (compute, solve) where

import Mylib.Prime (primeNumbers)

compute :: Int -> String
compute nth
    | nth <= 0 = error "The argument must be positive"
    | otherwise = show $ primeNumbers !! (pred nth)

solve :: String
solve = compute 10_001
